### PR TITLE
ipq40xx: add support for GL.iNet GL-AP1300

### PR DESCRIPTION
**Pull request by handongming**
Specifications:
SOC:        Qualcomm IPQ4018 (DAKOTA) ARM Quad-Core
RAM:        256 MiB
FLASH1:     4 MiB NOR
FLASH2:     128 MiB NAND
ETH:        Qualcomm QCA8075
WLAN1:      Qualcomm Atheros QCA4018 2.4GHz 802.11b/g/n 2x2
WLAN2:      Qualcomm Atheros QCA4018 5GHz 802.11n/ac W2 2x2
INPUT:      Reset
LED:        Power, Internet
UART1:      On board pin header near to LED (3.3V, TX, RX, GND), 3.3V without pin - 115200 8N1
OTHER:      On board with BLE module - by cp210x USB serial chip
            On board hareware watchdog with GPIO0 high to turn on, and GPIO4 for watchdog feed

Install via uboot tftp or uboot web failsafe.

By uboot tftp:
(IPQ40xx) # tftpboot 0x84000000 openwrt-ipq40xx-generic-glinet_gl-ap1300-squashfs-nand-factory.ubi
(IPQ40xx) # run lf

By uboot web failsafe:
Push the reset button for 10 seconds util the power led flash faster,
then use broswer to access http://192.168.1.1

Afterwards upgrade can use sysupgrade image.

Signed-off-by: Dongming Han <handongming@gl-inet.com>

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

### CODE REVIEW

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-gl-ap1300.dts [hunk 1]
```diff
@@ -0,0 +1,269 @@
+// SPDX-License-Identifier: GPL-2.0 OR MIT
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-gl-ap1300.dts:1]: That should be one of
GPL-2.0-only OR MIT
GPL-2.0-or-later OR MIT

**Comment by handongming** [comment]
Thanks for your review, updated.

**Comment by handongming** [comment]
Sorry I missed the uboot-envtools package, just updated and pushed, thanks.

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-gl-ap1300.dts [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+			gpios = <&tlmm 63 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "gl-ap1300:green:power";
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-gl-ap1300.dts:108]: model prefix "gl-ap1300:" can be dropped for LED labels now, just keep "green:power" etc.

File: target/linux/ipq40xx/patches-4.19/901-arm-boot-add-dts-files.patch [hunk 3]
```diff
@@ -10,7 +10,7 @@ Signed-off-by: John Crispin <john@phrozen.org>
 
 --- a/arch/arm/boot/dts/Makefile
 +++ b/arch/arm/boot/dts/Makefile
-@@ -785,11 +785,49 @@ dtb-$(CONFIG_ARCH_QCOM) += \
+@@ -785,11 +785,50 @@ dtb-$(CONFIG_ARCH_QCOM) += \
```
Comment (adschm) [inline target/linux/ipq40xx/patches-4.19/901-arm-boot-add-dts-files.patch:13]: This file can be removed, 4.19 has been dropped for this target.

**Comment by ynezz** [comment]
Please rebase.

**Comment by ynezz** [comment]
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git

**Pull request closed without merging**